app/views/handlers/individual_response.py

```diff
@@ -574,9 +598,11 @@ def handle_get(self):
         )
 
     def handle_post(self):
+        self._check_individual_response_count()
         if self.selected_option == self.confirm_option:
             self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
             self._publish_fulfilment_request()
+            self._update_individual_response_count()
```

Comment:
Checking the limit should only be relevant when they are requesting another IR, as in
when `self.selected_option == self.confirm_option`.
You could then swap the order of the calls to be:
```py
self._publish_fulfilment_request()
self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
```
where `_publish_fulfilment_request` handles all the related checks to do with limits and raise the expection as needed.